modin/core/storage_formats/base/query_compiler.py

```diff
@@ -169,6 +169,9 @@ def default_to_pandas(self, pandas_op, *args, **kwargs):
             warnings.filterwarnings("ignore", category=FutureWarning)
             result = pandas_op(try_cast_to_pandas(self), *args, **kwargs)
         if isinstance(result, (tuple, list)):
+            if "Series.tolist" in pandas_op.__name__:
+                # fast path: no need to iterate over the result from `tolist` function
```

Comment:
Does this branch prevent us from putting a list of values back to the engine storage?

Reply:
No, because a list element cannot be a dataframe or series (Placement in storage occurs only for them).